ipyvolume/pylab.py

```diff
@@ -675,67 +674,71 @@ def volshow(data, lighting=False, data_min=None, data_max=None,
-                    specular_coefficient = specular_coefficient,
-                    specular_exponent = specular_exponent,
-                    rendering_lighting = lighting)
-
-    vol._listen_to(fig)
+    data = np.moveaxis(data, -1, 0)  # for more convenient looping
+    for i, (subdata, color) in enumerate(zip(data, colors)):
+        tf = linear_transfer_function(color)
+        vol = ipv.Volume(data_original = subdata,
+                        tf=tf,
+                        data_min = data_min,
+                        data_max = data_max,
+                        show_min = data_min,
+                        show_max = data_max,
+                        extent_original = extent,
+                        data_max_shape = max_shape,
+                        ambient_coefficient = ambient_coefficient,
+                        diffuse_coefficient = diffuse_coefficient,
+                        specular_coefficient = specular_coefficient,
+                        specular_exponent = specular_exponent,
+                        rendering_lighting = lighting)
+
+        vol._listen_to(fig)
 
-    if controls:
-        widget_opacity_scale = ipywidgets.FloatLogSlider(base=10, min=-2, max=2,
-                                                     description="opacity")
-        widget_brightness = ipywidgets.FloatLogSlider(base=10, min=-1, max=1,
-                                                     description="brightness")
-        ipywidgets.jslink((vol, 'opacity_scale'), (widget_opacity_scale, 'value'))
-        ipywidgets.jslink((vol, 'brightness'), (widget_brightness, 'value'))
-        widgets_bottom = [ipywidgets.HBox([widget_opacity_scale, widget_brightness])]
-        current.container.children += tuple(widgets_bottom, )
-
-    fig.volumes = fig.volumes + [vol]
-
-    return vol
+        if controls:
+            widget_opacity_scale = ipywidgets.FloatLogSlider(base=10, min=-2, max=2,
+                                                         description="opacity")
+            widget_brightness = ipywidgets.FloatLogSlider(base=10, min=-1, max=1,
+                                                         description="brightness")
+            widget_colorpicker = ipywidgets.ColorPicker(value=color,
+                layout=ipywidgets.Layout(width='15%'))
+            ipywidgets.jslink((vol, 'opacity_scale'), (widget_opacity_scale, 'value'))
+            ipywidgets.jslink((vol, 'brightness'), (widget_brightness, 'value'))
+            def change_transfer_function(vol, color):
+                vol.tf = linear_transfer_function(color.new)
+            widget_colorpicker.observe(lambda x, vol=vol: change_transfer_function(vol, x), names='value')
+
+            widgets_bottom = [ipywidgets.HBox([widget_colorpicker, widget_opacity_scale, widget_brightness])]
+            current.container.children += tuple(widgets_bottom, )
+
+        fig.volumes = fig.volumes + [vol]
+
+    return fig
```

Comment:
I'm against returning figure, since it does not follow conventions/intuition.

Reply:
Return the full list of volumes, then? I'll do that.

Reply:
Well, check my above comment, why would volshow do this? Why not repeatedly call volshow?

Reply:
Btw, for a quick chat, https://gitter.im/maartenbreddels/ipyvolume may be useful, and we could also do a videochat if that speeds things up.